# **Group Project #2: Team Data Pioneers**

##### **Group Members**: Raquel Tartt, Kawika Keys, Oluwagbemisoke Soneye, Mehlik Thomas, and Maryshawnt Obilor
##### MATH 014- Data Science
##### Data Exploration and Wrangling
##### 04/07/2026

## **Topic**: *Data Exploration & Advanced Analytics*

## **Table of Contents**
#### ***Phase 1: Dataset Reconnaissance***
##### **Task 1: Data Set Information**
##### **Task 2: Basic Data Exploration**
##### **Task 3: Data Types & Structure**
##### **Task 4: Descriptive Summary**
##### **Task 5: Missing or Duplicate Data** 
#### ***Phase 2: Data Wrangling***
##### **Task 6: Add a New Column**
##### **Task 7: Filter the Data**
##### **Task 8: Unique Values and Categories**
#### ***Phase 3: Mission Debrief***
##### **Task 9: Initial Observations**

---

### **Task 01**: Data Set Information

#### Exploring Crime Patterns in Montgomery County Maryland

#### **Data Set Link**:https://catalog.data.gov/dataset/daily-arrests

In [61]:
import pandas as pd

In [62]:
df = pd.read_csv("Daily_Arrests.csv")

### **<u>Our Why:<u>** 

#### We selected the daily arrest dataset, which documents the arrests that occur in  Montgomery County, Maryland, because it provides detailed, real-world insight into patterns of crime and policing. With variables such as age, offense type, and location, the dataset allows for analysis of trends through the identification of crime hotspots and the most common types of offenses. Through this data set, the Data Pioneers can explore systemic patterns to contribute to better-informed criminal justice policies.  

#### The Daily Arrests dataset from Montgomery County provides regularly updated information on individuals processed through the county’s  Processing Unit. It includes variables such as age, offense type, and location, offering insight into patterns of crime and police activity. Since the data is updated frequently, it allows for timely analysis of trends, and since each record represents an individual arrest, it makes the dataset suitable for structured, quantitative analysis.


---

### **Task 02**: Basic Data Exploration 

#### First 5 entries 

In [63]:
df.head()

,LAST NAME,FIRST NAME,MIDDLE NAME,AGE,STREET,CITY,STATE,ARREST DATE,OFFENSE
0,NDLOVU,AYIBONGWE,AYIBONGWE,32,NO FIXED ADDRESS,ROCKVILLE,MD,04/03/2026,TRESPASS: PRIVATE PROPERTY
1,DOVE,TED,FRANKLIN,58,11600 Block Of NEBBLE ST,ROCKVILLE,MD,03/22/2026,TRESPASS: PRIVATE PROPERTY
2,WALKER,WILLIAM,HENRY,60,3500 Block Of ASTON MANOR CT,SILVER SPRING,MD,04/03/2026,LOADED HANDGUN IN VEHICLE
3,BECKWITH,JOVAN,ALLONTE,35,7500 Block Of WOODBINE,LAUREL,MD,04/03/2026,SEX OFF 3RD-CHILD <14
4,MORALES-GUZMAN,JENS,JOHATHAN,28,2900 Block Of HEWITT AVE,SILVER SPRING,MD,03/17/2026,(DRIVING/ATTEMPTING TO) DRIVE MOTOR VEH. ON HW...


#### Last 5 entries 

In [64]:
df.tail()

,LAST NAME,FIRST NAME,MIDDLE NAME,AGE,STREET,CITY,STATE,ARREST DATE,OFFENSE
427,GIRI,NIMESH,NaN,28,12100 Block Of SKYLARK RD,Clarksburg,MD,03/13/2026,SELL ALC BEV UNDER 21
428,SALGADO,DENNIS,JOSIAH,20,14900 Block Of LADYMEADE DR,SILVER SPRING,MD,03/13/2026,"PERSON DRIVING MOTOR VEH. ON (HWY., PUBLIC USE..."
429,PINEDA,LUIS,ADOLFO,39,3700 Block Of BEL PRE ROAD,SILVER SPRING,MD,03/13/2026,"(DRIVING, ATTEMPTING TO DRIVE) VEHICLE WHILE U..."
430,HAHN,JAMES,AUGUST,24,NO FIXED ADDRESS,ROCKVILLE,MD,03/13/2026,PEACE ORDER: FAIL TO COMPLY
431,DANDOR,KENNETH,SORIE,34,9500 Block Of Fen Hallow way,Montgomery Village,MD,03/14/2026,FRAUD-PER. IDENT. AVOID PROS


In [65]:
dimensions = df.shape

In [66]:
dimensions 

(432, 9)

#### The data set has 432 rows and 9 columns 

In [67]:
df.columns

Index(['LAST NAME', 'FIRST NAME', 'MIDDLE NAME', 'AGE', 'STREET', 'CITY',
       'STATE', 'ARREST DATE', 'OFFENSE'],
      dtype='object')

In [68]:
df.columns =['Last_Name', 'First_Name', 'Middle_Name', 'Age', 'Street', 'City', 
                    'State', 'Arrest_Date', 'Offense']

---

### **Task 03**: Data types & Structures

In [69]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 432 entries, 0 to 431
Data columns (total 9 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   Last_Name    432 non-null    object
 1   First_Name   432 non-null    object
 2   Middle_Name  358 non-null    object
 3   Age          432 non-null    int64 
 4   Street       432 non-null    object
 5   City         432 non-null    object
 6   State        431 non-null    object
 7   Arrest_Date  432 non-null    object
 8   Offense      432 non-null    object
dtypes: int64(1), object(8)
memory usage: 30.5+ KB


#### **Numerical Columns (Quantitative)**
#### Age: continuous/discrete numeric variable

#### **Categorical Columns (Qualitative)**
#### Full Name (first, middle, last) → identifier (nominal)
#### Address: categorical
#### Offense Type: categorical 
#### Date/Time → temporal 

#### **<u>Are there any data types that should be converted?<u>**
#### The dataset includes both numerical and categorical variables. Age is a numerical variable suitable for statistical analysis, while variables such as offense type, address, and name are categorical. Several columns require data type conversion to support analysis. For example, address information should be converted into zip codes or regions to better organize the data.

---

### **Task 04**: Descriptive Statistics 

In [70]:
df.describe(include='all')

,Last_Name,First_Name,Middle_Name,Age,Street,City,State,Arrest_Date,Offense
count,432,432,358,432.000000,432,432,431,432,432
unique,381,326,275,NaN,363,67,10,31,103
top,JUVENILE,JOSE,JUVENILE,NaN,NO FIXED ADDRESS,SILVER SPRING,MD,03/18/2026,ASSAULT-SECOND DEGREE
freq,9,13,9,NaN,61,103,395,24,57
mean,NaN,NaN,NaN,35.594907,NaN,NaN,NaN,NaN,NaN
std,NaN,NaN,NaN,12.390411,NaN,NaN,NaN,NaN,NaN
min,NaN,NaN,NaN,16.000000,NaN,NaN,NaN,NaN,NaN
25%,NaN,NaN,NaN,26.000000,NaN,NaN,NaN,NaN,NaN
50%,NaN,NaN,NaN,34.000000,NaN,NaN,NaN,NaN,NaN
75%,NaN,NaN,NaN,42.000000,NaN,NaN,NaN,NaN,NaN


#### **<u>What do the statistics tell us about quantitative data points?<u>**
#### **Top**: Most frequent response 
#### **Mean**: average age of individuals arrested (35.59)
#### **Standard deviation**: how spread out the ages are (12.39)
#### **Min**: youngest individual arrested (16)
#### **Max**: oldest individual arrested (75)
#### **Percentiles**: distribution (e.g., 75% of the individuals arrested are less than or equal to 42)

#### **<u>Are there any unusual or inconsistent values?<u>**

#### <u>Age Entries<u>

In [71]:
df[df['Age'] > 0]

,Last_Name,First_Name,Middle_Name,Age,Street,City,State,Arrest_Date,Offense
0,NDLOVU,AYIBONGWE,AYIBONGWE,32,NO FIXED ADDRESS,ROCKVILLE,MD,04/03/2026,TRESPASS: PRIVATE PROPERTY
1,DOVE,TED,FRANKLIN,58,11600 Block Of NEBBLE ST,ROCKVILLE,MD,03/22/2026,TRESPASS: PRIVATE PROPERTY
2,WALKER,WILLIAM,HENRY,60,3500 Block Of ASTON MANOR CT,SILVER SPRING,MD,04/03/2026,LOADED HANDGUN IN VEHICLE
3,BECKWITH,JOVAN,ALLONTE,35,7500 Block Of WOODBINE,LAUREL,MD,04/03/2026,SEX OFF 3RD-CHILD <14
4,MORALES-GUZMAN,JENS,JOHATHAN,28,2900 Block Of HEWITT AVE,SILVER SPRING,MD,03/17/2026,(DRIVING/ATTEMPTING TO) DRIVE MOTOR VEH. ON HW...
...,...,...,...,...,...,...,...,...,...
427,GIRI,NIMESH,NaN,28,12100 Block Of SKYLARK RD,Clarksburg,MD,03/13/2026,SELL ALC BEV UNDER 21
428,SALGADO,DENNIS,JOSIAH,20,14900 Block Of LADYMEADE DR,SILVER SPRING,MD,03/13/2026,"PERSON DRIVING MOTOR VEH. ON (HWY., PUBLIC USE..."
429,PINEDA,LUIS,ADOLFO,39,3700 Block Of BEL PRE ROAD,SILVER SPRING,MD,03/13/2026,"(DRIVING, ATTEMPTING TO DRIVE) VEHICLE WHILE U..."
430,HAHN,JAMES,AUGUST,24,NO FIXED ADDRESS,ROCKVILLE,MD,03/13/2026,PEACE ORDER: FAIL TO COMPLY


In [72]:
df[df['Age'] > 100]

,Last_Name,First_Name,Middle_Name,Age,Street,City,State,Arrest_Date,Offense


#### No Unusual Values in Age 

#### <u>City Entries<u>

In [73]:
df[df['City'] != df['City'].str.upper()]

,Last_Name,First_Name,Middle_Name,Age,Street,City,State,Arrest_Date,Offense
9,MITCHELL,MICHELLE,NaN,38,26600 Block Of Ridge Road,Damascus,MD,03/22/2026,ASSAULT-SECOND DEGREE
11,MOYLER,WILLIAM,EDWARD,26,4200 Block Of E Capitol,Washington D.C.,DC,03/27/2026,ASSAULT-FIRST DEGREE
17,JOHNSON,CHRISTIAN,DARWYN,36,100 Block Of 42nd street,Washington D.C.,MD,03/28/2026,DISORDERLY CONDUCT
23,AYALA,MAURICIO,NaN,46,9900 Block Of tambay ct,Montgomery Village,MD,03/28/2026,DANGEROUS WEAPON: CONCEAL
24,HARRIS,WAYNE,NaN,32,400 Block Of Douglas street,Washington D.C.,DC,03/18/2026,CDS POSS W/INT DIST-CANNABIS
...,...,...,...,...,...,...,...,...,...
399,REYES-ENAMORADO,CHRISTOFER,JOSHUA,19,3700 Block Of OLD BALTIMORE RD,Olney,MD,03/30/2026,ROBBERY
403,HOOD,STEPHEN,MARQUIS,35,100 Block Of UPSHIRE ST,Washington D.C.,DC,03/31/2026,KNOWINGLY DRIVING UNINSURED VEHICLE
422,RICKS,MAURICE,EVANDER,32,5300 Block Of James Place,Washington D.C.,DC,03/12/2026,MALICIOUS DESTRUCTION OF PROPERTY/VALUE LESS T...
427,GIRI,NIMESH,NaN,28,12100 Block Of SKYLARK RD,Clarksburg,MD,03/13/2026,SELL ALC BEV UNDER 21


In [74]:
(df['City'] != df['City'].str.upper()).sum()

np.int64(62)

In [75]:
df.loc[df['City'] != df['City'].str.upper(), 'City']

9                Damascus
11        Washington D.C.
17        Washington D.C.
23     Montgomery Village
24        Washington D.C.
              ...        
399                 Olney
403       Washington D.C.
422       Washington D.C.
427            Clarksburg
431    Montgomery Village
Name: City, Length: 62, dtype: object

In [76]:
df.loc[df['City'] != df['City'].str.upper(), 'City'].unique()

array(['Damascus', 'Washington D.C.', 'Montgomery Village', 'Clarksburg',
       'Fort Washington', 'Beltsville', 'Burtonsville',
       'Monterrey Village', 'Baltimore', 'Potomac', 'Adelphi',
       'Springdale', 'Boyds', 'Ijamsville', 'Lanham', 'Derwood', 'Olney'],
      dtype=object)

#### An inconsistency in text formatting was identified within the city column, where some entries were not in uppercase. In 62 rows, the listed city was written in a case that differed from the format of the majority of data entries.

#### <u>Offense Entries<u>

In [77]:
df[df['Offense'] != df['Offense'].str.upper()]

,Last_Name,First_Name,Middle_Name,Age,Street,City,State,Arrest_Date,Offense
18,MUNOZ MARQUEZ,GABRIEL,ARCANGEL,34,2700 Block Of LIVINGSTONE LN,VIENNA,VA,03/09/2026,Fugitive From Justice
20,ALDOSSARY,NAIF,JAMAL,32,3000 Block Of Van Dorn Rd,KNIGHTDALE,MD,04/01/2026,Fugitive From Justice
66,MEDINA CARDENAS,ERNESTO,SAMAEL,21,4900 Block Of powder mill rd,Beltsville,MD,03/27/2026,Prohibited Conduct/Fare Evasion
67,SANCHEZ MAGANA,WILLIAM,EDENILSON,28,8900 Block Of CENTER WAY RD,GAITHERSBURG,MD,03/27/2026,Prohibited Conduct/Fare Evasion
90,REYES JUARES,OVIDIO,ANTONIO,49,2600 Block Of Randolph Road,SILVER SPRING,MD,03/14/2026,Urinating / Defecating in Public
107,IRAHETA,JOSE,OSMAR,40,NO FIXED ADDRESS,SILVER SPRING,MD,04/02/2026,Disruptive Behavior-Pulic Facilities
109,HENDERSON,MARTIQUE,NaN,25,NO FIXED ADDRESS,GERMANTOWN,MD,04/03/2026,Fugitive From Justice
110,ARRINGTON,PAYTON,CHRISTIAN,23,4500 Block Of BESTER DR,ROCKVILLE,MD,03/12/2026,Fugitive From Justice
147,ANDERSON,JONATHAN,ROBERT,34,12400 Block Of WALNUT COVE CIRCLE,GERMANTOWN,MD,03/30/2026,Temporary detention by police officer of an in...
203,VASQUEZ CASCO,GERMAN,MAXIMINO,53,6400 Block Of 24TH AVE,HYATTSVILLE,MD,03/30/2026,Fugitive From Justice


In [78]:
(df['Offense'] != df['Offense'].str.upper()).sum()

np.int64(21)

In [79]:
df.loc[df['Offense'] != df['Offense'].str.upper(), 'Offense'].unique()

array(['Fugitive From Justice', 'Prohibited Conduct/Fare Evasion',
       'Urinating / Defecating in Public',
       'Disruptive Behavior-Pulic Facilities',
       'Temporary detention by police officer of an individual suspected of criminal behavior',
       'Urination and Defecation in Public'], dtype=object)

#### An inconsistency in text formatting was identified within the offense column, where some entries were not in uppercase. In 21 rows, the listed offense was written in a case that differed from the format of the majority of data entries.

---

### **Task 05**: Missing or Duplicate Values 

In [80]:
df.isnull().sum() 

Last_Name       0
First_Name      0
Middle_Name    74
Age             0
Street          0
City            0
State           1
Arrest_Date     0
Offense         0
dtype: int64

In [81]:
df.duplicated().sum()

np.int64(0)

#### There are 74 missing values within the "Middle_Name" column and 1 missing value within the "State" column. The missing "State" value is more impactful, as the value could include information that could support our objective of finding crime hotspots. The missing middle name values are not crucial to this data analysis, as such information does not assist with pattern recognition. Additonally, first, middle, and last names are personal identifiable information (PII), so for ethical and privacy purposes, they should be anonymized. 

---

### **Task 06**: Add a New Column 

In [82]:
df["Age_Group"] = 1

In [83]:
def categorize_age(age):
    if age < 18:
        return "Minor"
    elif age <= 25:
        return "Young Adult"
    elif age <= 40:
        return "Adult"
    else:
        return "Older Adult"

df['Age_Group'] = df['Age'].apply(categorize_age)
df.to_csv('Daily_Arrests_Updated.csv', index=False)

In [84]:
df.head(10)

,Last_Name,First_Name,Middle_Name,Age,Street,City,State,Arrest_Date,Offense,Age_Group
0,NDLOVU,AYIBONGWE,AYIBONGWE,32,NO FIXED ADDRESS,ROCKVILLE,MD,04/03/2026,TRESPASS: PRIVATE PROPERTY,Adult
1,DOVE,TED,FRANKLIN,58,11600 Block Of NEBBLE ST,ROCKVILLE,MD,03/22/2026,TRESPASS: PRIVATE PROPERTY,Older Adult
2,WALKER,WILLIAM,HENRY,60,3500 Block Of ASTON MANOR CT,SILVER SPRING,MD,04/03/2026,LOADED HANDGUN IN VEHICLE,Older Adult
3,BECKWITH,JOVAN,ALLONTE,35,7500 Block Of WOODBINE,LAUREL,MD,04/03/2026,SEX OFF 3RD-CHILD <14,Adult
4,MORALES-GUZMAN,JENS,JOHATHAN,28,2900 Block Of HEWITT AVE,SILVER SPRING,MD,03/17/2026,(DRIVING/ATTEMPTING TO) DRIVE MOTOR VEH. ON HW...,Adult
5,JUVENILE,JUVENILE,JUVENILE,16,14100 Block Of GRAND PRE RD,SILVER SPRING,MD,03/09/2026,ASSAULT-FIRST DEGREE,Minor
6,PEACOCK,EDWIN,ADEYINKA,48,NO FIXED ADDRESS,ROCKVILLE,MD,04/01/2026,MALICIOUS BURNING/SECOND DEGREE,Older Adult
7,REYES RIVERA,JOSE,HERMAN,41,7500 Block Of brink rd,GAITHERSBURG,MD,03/15/2026,ATTEMPT-SECOND DEGREE RAPE,Older Adult
8,SESAY,PATIENCE,NaN,55,NO FIXED ADDRESS,SILVER SPRING,MD,03/10/2026,TRESPASS: PRIVATE PROPERTY,Older Adult
9,MITCHELL,MICHELLE,NaN,38,26600 Block Of Ridge Road,Damascus,MD,03/22/2026,ASSAULT-SECOND DEGREE,Adult


In [85]:
df.tail(10)

,Last_Name,First_Name,Middle_Name,Age,Street,City,State,Arrest_Date,Offense,Age_Group
422,RICKS,MAURICE,EVANDER,32,5300 Block Of James Place,Washington D.C.,DC,03/12/2026,MALICIOUS DESTRUCTION OF PROPERTY/VALUE LESS T...,Adult
423,AREVALO CRUZ,AMY,SARAI,18,20000 Block Of APPLEDOWRE CIR,GERMANTOWN,MD,03/13/2026,ASSAULT-SECOND DEGREE,Young Adult
424,DASILVA-BAPTISTA,IRVIN,NaN,18,800 Block Of HIDDEN MARSH ST,GAITHERSBURG,MD,03/10/2026,DANGEROUS DEADLY WEAPON: ON SCHOOL PROPERTY,Young Adult
425,TURNER,EUGENE,NaN,30,8500 Block Of FLOWER AVE,TAKOMA PARK,MD,03/13/2026,FRAUD-PER. IDENT. AVOID PROS,Adult
426,REDMOND,ILYA,GREGORY,29,9800 Block Of WHISKEY RUN,LAUREL,MD,03/13/2026,SEX ABUSE MINOR-CONTINUING COURSE OF CONDUCT,Adult
427,GIRI,NIMESH,NaN,28,12100 Block Of SKYLARK RD,Clarksburg,MD,03/13/2026,SELL ALC BEV UNDER 21,Adult
428,SALGADO,DENNIS,JOSIAH,20,14900 Block Of LADYMEADE DR,SILVER SPRING,MD,03/13/2026,"PERSON DRIVING MOTOR VEH. ON (HWY., PUBLIC USE...",Young Adult
429,PINEDA,LUIS,ADOLFO,39,3700 Block Of BEL PRE ROAD,SILVER SPRING,MD,03/13/2026,"(DRIVING, ATTEMPTING TO DRIVE) VEHICLE WHILE U...",Adult
430,HAHN,JAMES,AUGUST,24,NO FIXED ADDRESS,ROCKVILLE,MD,03/13/2026,PEACE ORDER: FAIL TO COMPLY,Young Adult
431,DANDOR,KENNETH,SORIE,34,9500 Block Of Fen Hallow way,Montgomery Village,MD,03/14/2026,FRAUD-PER. IDENT. AVOID PROS,Adult


#### This new column groups individuals into age categories (Minor, Young Adult, Adult, and Older Adult), which makes it easier to analyze arrest patterns and observe the relationship between age and criminal activity. Additonally, age grouping aligns more closely with how policies and interventions are designed. For example, programs for minors, young adults, or older adults differ, so grouping ages makes the data more useful for real-world decision-making.

---

### **Task 07**: Filter the Data

#### <u>FILTER 1: Under 25 Years Old<u>

In [86]:
filter1 = df.query('Age <= 25')

In [87]:
filter1

,Last_Name,First_Name,Middle_Name,Age,Street,City,State,Arrest_Date,Offense,Age_Group
5,JUVENILE,JUVENILE,JUVENILE,16,14100 Block Of GRAND PRE RD,SILVER SPRING,MD,03/09/2026,ASSAULT-FIRST DEGREE,Minor
25,JACKSON,CYDNEY,OLIVIA,25,8300 Block Of BRODERICK CIR,GAITHERSBURG,MD,03/28/2026,ASSAULT-SECOND DEGREE,Young Adult
32,RODRIGUEZ ROMERO,JOVANY,JOSUE,18,700 Block Of QUINCE ORCHARD DRIVE,GAITHERSBURG,MD,03/28/2026,"THEFT: $100 TO UNDER $1,500",Young Adult
33,WOULLARD,JAYLIN,RICHARD,24,NO FIXED ADDRESS,ROCKVILLE,MD,03/22/2026,"THEFT: $100 TO UNDER $1,500",Young Adult
43,GROOM,JASMINE,ALIYAH,23,0 Block Of longworth ct,GAITHERSBURG,MD,03/10/2026,KNOWINGLY DRIVING UNINSURED VEHICLE,Young Adult
...,...,...,...,...,...,...,...,...,...,...
413,SPEER,SANIYA,NaN,21,5400 Block Of MARINELLI,ROCKVILLE,MD,04/01/2026,"THEFT: $100 TO UNDER $1,500",Young Adult
423,AREVALO CRUZ,AMY,SARAI,18,20000 Block Of APPLEDOWRE CIR,GERMANTOWN,MD,03/13/2026,ASSAULT-SECOND DEGREE,Young Adult
424,DASILVA-BAPTISTA,IRVIN,NaN,18,800 Block Of HIDDEN MARSH ST,GAITHERSBURG,MD,03/10/2026,DANGEROUS DEADLY WEAPON: ON SCHOOL PROPERTY,Young Adult
428,SALGADO,DENNIS,JOSIAH,20,14900 Block Of LADYMEADE DR,SILVER SPRING,MD,03/13/2026,"PERSON DRIVING MOTOR VEH. ON (HWY., PUBLIC USE...",Young Adult


#### This group captures youth and young adults, who are often at higher risk of involvement in criminal behavior. Analyzing this subset can reveal patterns in early criminal behavior, helping identify opportunities for early intervention, education/extracurricular programs, youth engagement-focused policy work, and crime prevention strategies.

#### <u>FILTER 2: City and Dangerous Drug Related Arrest<u>

In [88]:
df = pd.read_csv('Daily_Arrests_Updated.csv')

In [89]:
filter2 = df[
    (df['State'] == 'MD') & 
    (df['Offense'].str.contains('DANGEROUS WEAPON', case=False, na=False))
]
filter2

,Last_Name,First_Name,Middle_Name,Age,Street,City,State,Arrest_Date,Offense,Age_Group
23,AYALA,MAURICIO,NaN,46,9900 Block Of tambay ct,Montgomery Village,MD,03/28/2026,DANGEROUS WEAPON: CONCEAL,Older Adult
35,BODDIE,FREDERICK,JACOB,40,700 Block Of Roeder rd,SILVER SPRING,MD,04/02/2026,DANGEROUS WEAPON: CONCEAL,Adult
105,FLORES,JOSE,ALEXANDER,31,NO FIXED ADDRESS,SILVER SPRING,MD,04/02/2026,DANGEROUS WEAPON: CONCEAL,Adult
389,MEJOR HERNANDEZ,JOSE,MARTIN,31,2000 Block Of GEORGIAN WOODS PLACE,SILVER SPRING,MD,03/12/2026,DANGEROUS WEAPON: CONCEAL,Adult
402,WHITE,ALAN,DESMOND,26,200 Block Of Lake Coventry Dr,FREDERICK,MD,03/26/2026,DANGEROUS WEAPON: CONCEAL,Adult


#### This filter isolates more serious/violent offenses within a specific geographic area. It can help identify regional crime patterns and inform policy decisions around weapon-related laws in Maryland.

#### <u>FILTER 3: Relationship between Demographic, Geographic, and Economic Factors and Theft Related Arrest <u>

In [90]:
filter3 = df.query(
    'Age_Group == "Older Adult" and Street == "NO FIXED ADDRESS" and City == "ROCKVILLE" and Offense.str.contains("THEFT", case=False)',
    engine='python'
)

In [91]:
filter3

,Last_Name,First_Name,Middle_Name,Age,Street,City,State,Arrest_Date,Offense,Age_Group
21,STEVENSON,TRINA,MARIE,59,NO FIXED ADDRESS,ROCKVILLE,MD,03/09/2026,"THEFT: $100 TO UNDER $1,500",Older Adult
78,BREEN,KRISTEN,MARY,47,NO FIXED ADDRESS,ROCKVILLE,MD,04/01/2026,"THEFT: $100 TO UNDER $1,500",Older Adult
135,JOHNSON,WARREN,ANTONIO,71,NO FIXED ADDRESS,ROCKVILLE,MD,03/27/2026,THEFT LESS THAN $100.00,Older Adult
281,JOBE,BABOUCARR,NaN,41,NO FIXED ADDRESS,ROCKVILLE,MD,03/17/2026,"THEFT: $100 TO UNDER $1,500",Older Adult
289,PLUMMER,EARL,NaN,57,NO FIXED ADDRESS,ROCKVILLE,MD,03/17/2026,"THEFT: $100 TO UNDER $1,500",Older Adult


#### This subset focuses on older adults in Rockville who have experienced housing instability and were arrested for theft. It illustrates how age-related economic vulnerability and regional factors may intersect to influence criminal behavior patterns. This data can be examined to explore how challenges such as fixed income limitations, unemployment, and decreased social support might motivate economically driven offenses like theft. Additionally, the localized nature of this analysis can help inform policy decisions within the Rockville area.

### **Filter 4**: No Fixed Address 

In [92]:
filter4 = df[
    (df['Street'] == 'NO FIXED ADDRESS')]
filter4

,Last_Name,First_Name,Middle_Name,Age,Street,City,State,Arrest_Date,Offense,Age_Group
0,NDLOVU,AYIBONGWE,AYIBONGWE,32,NO FIXED ADDRESS,ROCKVILLE,MD,04/03/2026,TRESPASS: PRIVATE PROPERTY,Adult
6,PEACOCK,EDWIN,ADEYINKA,48,NO FIXED ADDRESS,ROCKVILLE,MD,04/01/2026,MALICIOUS BURNING/SECOND DEGREE,Older Adult
8,SESAY,PATIENCE,NaN,55,NO FIXED ADDRESS,SILVER SPRING,MD,03/10/2026,TRESPASS: PRIVATE PROPERTY,Older Adult
21,STEVENSON,TRINA,MARIE,59,NO FIXED ADDRESS,ROCKVILLE,MD,03/09/2026,"THEFT: $100 TO UNDER $1,500",Older Adult
33,WOULLARD,JAYLIN,RICHARD,24,NO FIXED ADDRESS,ROCKVILLE,MD,03/22/2026,"THEFT: $100 TO UNDER $1,500",Young Adult
...,...,...,...,...,...,...,...,...,...,...
401,HILLIARD,ARLYN,RODRIGUEZ,56,NO FIXED ADDRESS,ROCKVILLE,MD,04/01/2026,CDS:POSSESS PARAPHERNALIA,Older Adult
406,LEAVITT,STUART,THOMAS,68,NO FIXED ADDRESS,SILVER SPRING,MD,03/26/2026,DISORDERLY CONDUCT,Older Adult
408,ALANIZ,CHRISTIAN,ALEXANDER,25,NO FIXED ADDRESS,GERMANTOWN,MD,03/26/2026,"THEFT: $100 TO UNDER $1,500",Young Adult
419,HANNA,JALEN,SOLOMON,26,NO FIXED ADDRESS,ROCKVILLE,MD,04/01/2026,SEX OFF 4TH DEGREE-SEX CONTACT,Adult


#### This filter demonstrates the frequent appearance of “NO FIXED ADDRESS” as a listed street, suggesting that a portion of individuals being arrested may be experiencing housing instability. This subset can be used to examine the relationship between housing instability and criminal behavior, as these variables may indicate bidirectional causation. 

---

### **Task 08**: Unique Values and Categories

In [93]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 432 entries, 0 to 431
Data columns (total 10 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   Last_Name    432 non-null    object
 1   First_Name   432 non-null    object
 2   Middle_Name  358 non-null    object
 3   Age          432 non-null    int64 
 4   Street       432 non-null    object
 5   City         432 non-null    object
 6   State        431 non-null    object
 7   Arrest_Date  432 non-null    object
 8   Offense      432 non-null    object
 9   Age_Group    432 non-null    object
dtypes: int64(1), object(9)
memory usage: 33.9+ KB


In [94]:
df['City'].unique()

array(['ROCKVILLE', 'SILVER SPRING', 'LAUREL', 'GAITHERSBURG', 'Damascus',
       'BLADENSBURG', 'Washington D.C.', 'HYATTSVILLE', 'VIENNA',
       'KNIGHTDALE', 'SEVERN', 'Montgomery Village', 'BETHESDA',
       'NEW YORK', 'ABINGDON', 'RADFORD', 'FALLS CHURCH', 'GERMANTOWN',
       'Clarksburg', 'FREDERICK', 'Fort Washington', 'DISTRICT HEIGHTS',
       'KENSINGTON', 'Beltsville', 'Burtonsville', 'Monterrey Village',
       'WOODBRIDGE', 'WALDORF', 'UPPER MARLBORO', 'CHEVY CHASE',
       'Baltimore', 'CLEVELAND', 'Potomac', 'ASHBURN', 'WHEATON',
       'BROOKEVILLE', 'ALLENTOWN', 'TAKOMA PARK', 'Adelphi',
       'CHARLOTTESVILLE', 'Springdale', 'HERNDON', 'BALTIMORE',
       'DUMFRIES', 'GREENBELT', 'BRONX', 'BOWIE', 'WASHINGTON', 'Boyds',
       'ODENTON', 'FREDERICKSBURG', 'CAPITOL HEIGHTS', 'ATLANTA',
       'Ijamsville', 'Lanham', 'BROOKLYN', 'LEESBURG', 'ASPEN HILL',
       'ARLINGTON', 'MIDDLETOWN', 'Derwood', 'NEW CARROLLTON',
       'BELTSVILLE', 'HAGERSTOWN', 'RIVERDALE', 'C

In [95]:
df['City'].value_counts()

City
SILVER SPRING      103
ROCKVILLE           79
GAITHERSBURG        51
GERMANTOWN          40
Washington D.C.     21
                  ... 
Adelphi              1
Springdale           1
HERNDON              1
DUMFRIES             1
Olney                1
Name: count, Length: 67, dtype: int64

In [96]:
df['Arrest_Date'].unique()

array(['04/03/2026', '03/22/2026', '03/17/2026', '03/09/2026',
       '04/01/2026', '03/15/2026', '03/10/2026', '03/27/2026',
       '03/28/2026', '03/18/2026', '04/02/2026', '03/06/2026',
       '03/08/2026', '03/14/2026', '03/16/2026', '03/13/2026',
       '03/12/2026', '03/07/2026', '03/19/2026', '03/23/2026',
       '03/31/2026', '03/29/2026', '03/26/2026', '03/30/2026',
       '04/04/2026', '03/21/2026', '03/24/2026', '03/20/2026',
       '03/11/2026', '04/05/2026', '03/25/2026'], dtype=object)

In [97]:
df['Arrest_Date'].value_counts()

Arrest_Date
03/18/2026    24
03/27/2026    23
03/10/2026    22
03/17/2026    22
03/11/2026    21
03/30/2026    20
03/25/2026    19
04/01/2026    19
03/16/2026    18
03/07/2026    18
03/31/2026    17
03/19/2026    17
03/24/2026    16
03/26/2026    16
03/12/2026    16
03/20/2026    14
03/23/2026    14
04/02/2026    14
03/21/2026    12
03/13/2026    12
03/28/2026    12
03/14/2026    11
04/03/2026    11
03/09/2026    10
04/04/2026     7
03/22/2026     6
03/15/2026     6
03/08/2026     5
04/05/2026     5
03/29/2026     4
03/06/2026     1
Name: count, dtype: int64

In [98]:
categorical_cols = df.select_dtypes(include='object').columns

for col in categorical_cols:
    print(f"\nColumn: {col}")
    print(df[col].value_counts())


Column: Last_Name
Last_Name
JUVENILE          9
JOHNSON           4
BROWN             4
JONES             3
JACKSON           3
                 ..
SAMUEL III        1
ESPINAL ZAVALA    1
SALADO            1
RAMIREZ-LOPEZ     1
DANDOR            1
Name: count, Length: 381, dtype: int64

Column: First_Name
First_Name
JOSE         13
JUVENILE      9
ROBERT        6
LUIS          6
EDWIN         5
             ..
JAHI          1
ELLINGTON     1
JECZIEL       1
HADI          1
DENNIS        1
Name: count, Length: 326, dtype: int64

Column: Middle_Name
Middle_Name
JUVENILE        9
ALEXANDER       8
ANTONIO         7
ROBERT          5
EDWARD          5
               ..
MCKINLEY        1
DIEGO           1
NASIR HASSAN    1
BECHET          1
SORIE           1
Name: count, Length: 275, dtype: int64

Column: Street
Street
NO FIXED ADDRESS                    61
8100 Block Of GEORGIA AVE            5
400 Block Of SUMMIT AVE              2
4900 Block Of Powder Mill Rd         2
14200 Block Of WE

#### These types of categorical data such as, City, Arrest_Date, Offense, and Age_Group can help us to better understand and visualize the data. For instance, using the categories of City and Arrest_Date, we would be able to make a heat map of when and where exactly the crimes took place. Then, by adding in the Offense category, we would be able to see what exactly happened in relation to the place and time.

---

### **Task 09**: Initial Observations

#### Exploring the Montgomery County Daily Arrests dataset revealed several compelling patterns and raised important questions. One of the most striking findings was the frequent appearance of “NO FIXED ADDRESS” as a listed street, suggesting that a notable portion of individuals being arrested may be experiencing housing instability, especially in Rockville, MD. Theft emerged as one of the most common offense types across all age groups, particularly among individuals who noted having no fixed address, indicating that many arrests may be economically motivated rather than violent in nature. Notably, older adults (45+) appear disproportionately in theft-related arrests concentrated in Rockville, highlighting how age-related factors such as limited income, unemployment, or reduced social support may influence criminal behavior. Additionally, theft and assault emerged as the most frequent offenses in minor and young adult populations, which can meaningfully drive targeted reforms, school-based prevention programs, and youth engagement strategies.

#### The dataset also contained missing values, primarily in the Middle Name column, which, while not critical to analysis, reflects inconsistencies in data entry practices. Additionally, the wide variation in offense descriptions, such as multiple tiers of theft based on dollar amounts, introduced challenges in categorization and required careful interpretation during analysis.

#### Overall, these patterns suggest that arrests in Montgomery County may be shaped not only by individual behavior but also by broader social and economic conditions. Furthermore, the geographic variation across cities like Rockville, Silver Spring, and Laurel raises important questions about whether differences in arrest patterns reflect underlying community conditions or variations in policing practices.











